src/string_enums/formula_variant.hpp

```diff
@@ -28,7 +28,6 @@ struct formula_variant_defines
 	static constexpr const char* const type_string = "string";
 	static constexpr const char* const type_map = "map";
 
-	enum class type { TYPE_NULL, TYPE_INT, TYPE_DECIMAL, TYPE_OBJECT, TYPE_LIST, TYPE_STRING, TYPE_MAP, ENUM_MAX };
-	static constexpr std::array<const char*, static_cast<int>(type::ENUM_MAX)> values{ type_null, type_int, type_decimal, type_object, type_list, type_string, type_map };
+	ENUM_AND_ARRAY(type_null, type_int, type_decimal, type_object, type_list, type_string, type_map)
```

Comment:
You know, I don't think there's any reason these all need to start with `type_`, is there? I imagine that was a relic of it once being an unscoped enum?

Reply:
I also can't have a variable named just `int`, etc.

Reply:
That's not an issue, because "int" is short for "integer" in the first place, so you can just de-shorten it.

Reply:
I'd prefer for them to either all have the prefix or for none of them to have the prefix just for consistency.

Reply:
Yeah, that goes without saying. I don't _expect_ `null` to be a problem, but I wouldn't put it past some obscure header on some obscure OS (or even on Windows) to define `null` as a macro for a null pointer. If that _did_ happen though, we could just call it something else, like `nil` or `nothing` or something like that.

Reply:
Oh, whoops, I was getting it mixed up with Java where it's just `null` instead of `nullptr`. So yeah, that (hopefully) wouldn't be a problem.